src/components/cryptoDollar/cryptoDollarReducer.js

```diff
@@ -37,13 +37,15 @@ export default function (state = initialState, action) {
         ...state,
         contractState: {
           loading: true,
-          data: null
+          data: null,
+          ...state.contractState
```

Comment:
All the rest of the spreads that you added in this file look good and are placed correctly, but this one should be moved up two lines like the other ones below